app/controllers/registrations_controller.rb

```diff
@@ -4,6 +4,13 @@ class RegistrationsController < Devise::RegistrationsController
   before_action :load_game, :load_message_count
   before_action :check_captcha, only: %i[create]
   before_action :prevent_action_after_game, only: %i[new create]
+  before_action :prevent_action_if_registration_closed, only: %i[new create]
```

Comment:
These can be merged into 1 line:

`before_action :prevent_action_after_game, :prevent_action_if_registration_closed, only: %i[new create]`